src/adapter/graphql/tests/tests/test_gql_metadata_chain.rs

```diff
@@ -403,8 +405,10 @@ async fn metadata_update_readme_new() {
 
     assert_result(res, "CommitResultSuccess");
 
+    let dataset = dataset_repo.get_dataset(&dataset_ref).await.unwrap();
```

Comment:
Make a comment that we need to start a new transaction.
Maybe also scope the parts of the tests with separated blocks, so that the lifetime of the dataset/cache is clear.